### DISCUSSION TITLE
@stream and @defer directives

### DISCUSSION THREAD

**Discussion by MarcBT** [MichalLytek/type-graphql#892] [Ideas]
Hello 👋!

If I'm not wrong, the [`@stream` and `@defer` directives](https://graphql.org/blog/2020-12-08-improving-latency-with-defer-and-stream-directives) are not available with `type-graphql`.

I'm curious to know if there are plans to make them available in `type-graphql`.

_Apologies if this is not the right place to ask this kind of question_
_And thanks @MichalLytek for your hard work to build and maintain this project!_

**Comment by MichalLytek** [comment]
From my understanding, there's nothing to do on TypeGraphQL side.

`@stream` and `@defer` directives are placed on the client side (and TypeGraphQL has support for server side directives too). 

As TypeGraphQL is built upon the `graphql-js`, we just need to wait for it to stabilize and become a part of the official spec.

Also, looks like there's no new fields or API for declaring the schema on the `graphql-js` side. The only thing that might need adjustments is the support for returning async iterable from `resolve` in order to work with streams, as right now resolvers are wrapped in async/await in order to apply middlewares and other layers.

**Reply by FezVrasta** [reply]
Hi @MichalLytek, when I try to use @defer with Yoga, @graphql-yoga/plugin-defer-stream, and Relay I get `Unknown directive 'defer'`.

I think all is missing is to define the `@defer` directive, how can I do it with typegraphql?

**Reply by MichalLytek** [reply]
register in `directives` buildSchema option, like all other directives?

**Reply by FezVrasta** [reply]
Ah yes it was it 😅 thanks
